exonum/src/blockchain/service.rs

```diff
@@ -63,6 +65,204 @@ pub trait Transaction: Message + ExonumJson + 'static {
     fn execute(&self, fork: &mut Fork);
 }
 
+/// `TransactionSet` trait describes a type which is an `enum` of several transactions.
+/// The implementation of this trait is generated automatically by the `transactions!`
+/// macro.
+pub trait TransactionSet
```

Comment:
A bit nitpick-y, but: why is the trait called `*Set`? I'd say that it describes a [tagged union](https://en.wikipedia.org/wiki/Tagged_union) (with variants tagged by `message_id`), or `enum` in Rust terms.

Reply:
I am not too thrilled by `Set` myself, but I don't know of a better name :( 

TransactionSet can represent either of a *set* of transactions, though, this is a type-level, and not value-level set.